File: TypeScript/11.盛最多水的容器.ts
function maxArea(height: number[]): number {
    let left = 0, right = height.length-1;
    let ans = 0;
    while (left < right) {
        ans = Math.max(ans, (right - left) * Math.min(height[left], height[right]));
        if (height[left] < height[right]) {
            ++left;
        } else {
            --right;
        }
    }
    return ans;
};

File: Golang/11.盛最多水的容器.go
package leetcode

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}

func maxArea(height []int) int {
	left, right := 0, len(height)-1
	ans := 0
	for left < right {
		ans = max(ans, (right-left)*min(height[left], height[right]))
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return ans
}
